### ISSUE TITLE
prefect cloud CLI command ignores --profile argument

### ISSUE THREAD

**Issue by LeeMendelowitz** [PrefectHQ/prefect#7529]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

The `prefect cloud` CLI command ignores the `--profile`argument. This makes it tricky to use the CLI to configure two profiles to use two different workspaces.

I would expect that:

```bash
prefect profile create test
prefect --profile test cloud login --key xxx_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX -w "my-org/test"

prefect profile create production
prefect --profile production cloud login --key xxx_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX -w "my-org/production"
```

should configure the `test` and `production` profiles with the provided API key and set the provided workspaces, but that's not what happens. The first call to `prefect cloud login` asks for a profile name, but fails when the profile already exists:

```
Creating a profile for this Prefect Cloud login. Please specify a profile name: test
Profile 'test' already exists.
```

As a result, it seems like the easiest way to configure the CLI to use two different profiles with two different workspaces is to edit the `~/.prefect/profiles.toml` directly.

### Reproduction

```python3
# Start with an empty ~/.prefect/profiles.toml
prefect profile create test
prefect profile create production
prefect --profile test cloud login --key xxx_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX -w "my-org/test"
prefect --profile production cloud login --key xxx_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX -w "my-org/production"
```

### Versions

```Text
Version:             2.6.7
API version:         0.8.3
Python version:      3.10.8
Git commit:          9074c158
Built:               Thu, Nov 10, 2022 1:07 PM
OS/Arch:             darwin/arm64
Profile:             default
Server type:         ephemeral
Server:
  Database:          sqlite
  SQLite version:    3.37.0
```

**Comment by madkinsz** [comment]
Thanks for the well-written report! This should be addressed by #7334

**Comment by serinamarie** [comment]
I'm going to close this as completed. Please feel free to open a new issue if that is not the case.
